src/test/java/org/broadinstitute/hellbender/utils/config/ConfigIntegrationTest.java

```diff
@@ -0,0 +1,126 @@
+package org.broadinstitute.hellbender.utils.config;
+
+import org.aeonbits.owner.Config;
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.barclay.argparser.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.CommandLineProgramTest;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.cmdline.TestProgramGroup;
+import org.broadinstitute.hellbender.engine.GATKTool;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.utils.ClassUtils;
+import org.broadinstitute.hellbender.utils.test.ArgumentsBuilder;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.lang.reflect.Method;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.Properties;
+
+/**
+ * Integration test
+ * Created by jonn on 12/7/17.
+ */
+public class ConfigIntegrationTest extends CommandLineProgramTest {
+
+    @CommandLineProgramProperties(
+            summary = "Dummy empty command line that requires a reference .",
+            oneLineSummary = "empty class",
+            programGroup = TestProgramGroup.class
+    )
+    public static class DummyGatkTool extends GATKTool {
+
+        @Argument(
+                doc = "Output file with only the configuration settings in it.",
+                fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+                shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME
+        )
+        private File outputFilePath;
+
+        @Override
+        public void traverse() {
+            final GATKConfig gatkConfig = ConfigFactory.getInstance().getOrCreate(GATKConfig.class);
+            ConfigFactory.dumpConfigSettings( gatkConfig, outputFilePath.toPath() );
+        }
+    }
+
+    @Override
+    public String getTestedClassName() {
+        return DummyGatkTool.class.getSimpleName();
+    }
+
+    @DataProvider
+    private Object[][] provideForTestToolWithConfigValidation() {
+        return new Object[][] {
+                { getSafeNonExistentPath("nonExistentConfig.config") },
```

Comment:
There should be 2 test cases here: one that runs GATK without a config argument at all, and a second that runs with a config argument pointing to a file that overrides at least one default config setting.

Reply:
Right - I meant to go back and fix this.  Will do now.